app/assets/javascripts/backbone/views/timeline_select_steps_view.coffee

```diff
@@ -9,44 +9,40 @@ class MS.Views.TimelineSelectStepsView extends Backbone.View
     'click #uncheck-all-ctrl': 'uncheckAll'
     'click #check-completed-ctrl': 'checkCompleted'
     'click #check-incomplete-ctrl': 'checkIncomplete'
- 
-  checkAll: (e) ->
-    $control = $(e.currentTarget)
-    $inputs = $control.closest('.timeline').find('.select-step')
-    @.checkItems($inputs)
 
-    $masterCheckbox = $('#choose-all')
-    @.checkItems($masterCheckbox)
+  checkAll: () ->
+    @.toggleAll(true)
 
-    @.rememberChecked(e)
+  uncheckAll: () ->
+    @.toggleAll(false)
 
-  uncheckAll: (e) ->
-    $control = $(e.currentTarget)
-    $inputs = $control.closest('.timeline').find('.select-step')
-    @.uncheckItems($inputs)
+  checkCompleted: () ->
+    $inputs = @$el.find('.completed-item')
+    @.toggleSubset(true, $inputs)
 
-    $masterCheckbox = $('#choose-all')
-    @.uncheckItems($masterCheckbox)
+  checkIncomplete: () ->
+    $inputs = @$el.find('.incomplete-item')
+    @.toggleSubset(true, $inputs)
```

Comment:
Please make this and the above into one-liners. No need for the extra verbosity.